### PR TITLE
Implemented `Feature::set_field_null`

### LINKED ISSUE #427: FR: add wrapper for OGR_F_SetFieldNull
All of the "get" field functions on Feature return options, so I can know if the value is already null. But I don't see anything in the rust API for setting a field to None that shouldn't have a value, and `OGR_F_UnsetField` is only referenced in the pre-built bindings on a sourcecode search here.

My work-around right now is to use `feature.set_field_double(<field_name>,f64::NAN)` but I've only had to do this for double fields so far, so I don't know if this will work for integer, or if it will work in other drivers than geopackage.

### PR DESCRIPTION
**Pull request by metasim**
- [X] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [X] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Closes #427 
Credit to @nms-scribe

### CODE REVIEW

File: src/vector/feature.rs [hunk 1]
```diff
@@ -911,9 +926,32 @@ pub fn field_type_to_name(ty: OGRFieldType::Type) -> String {
...
+        assert_eq!(field_type_to_name(OGRFieldType::OFTReal), "Real");
+        // We don't care what it returns when passed an invalid value, just that it doesn't crash.
+        field_type_to_name(4372521);
+    }
+
+    #[test]
+    fn test_field_set_null() {
+        let ds = Dataset::open(fixture("roads.geojson")).unwrap();
+
+        for mut layer in ds.layers() {
```
Comment (lnicola) [inline src/vector/feature.rs:946]: I think we can limit this to a single layer and feature.

**Comment by lnicola** [comment]
I was too afraid to check in #427, but still have no idea. Does anyone know the difference between `OGR_F_SetFieldNull` and `OGR_F_UnsetField`?

Does the latter actually remove the field value in formats like GeoJSON?

**Comment by metasim** [comment]
[Docs](https://gdal.org/api/vector_c_api.html): 

* `OGR_F_SetFieldNull`: Clear a field, marking it as null.
* `OGR_F_UnsetField`: Clear a field, marking it as unset.

@lnicola Yeah, I don't understand the semantics either.

Maybe we just mark this as "draft" and come back to it later?

**Comment by lnicola** [comment]
No, I would have merged it, but unsetting all fields of all features in all layers seems a bit excessive. But it's only 21 features, so feel free to merge :slightly_smiling_face:.

**Comment by metasim** [comment]
I simplified the test.

**Pull request merged**